x-pack/plugins/canvas/common/lib/fonts.ts

```diff
@@ -4,56 +4,106 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-export const americanTypewriter = {
+export type FontValue = typeof fonts[number]['value'];
+export type FontLabel = typeof fonts[number]['label'];
+
+export interface Font {
+  label: FontLabel;
+  value: FontValue;
+}
+
+/**
+ * This function allows one to create a strongly-typed font for inclusion in
+ * the font collection.  As a result, the values and labels are known to the
+ * type system, preventing one from specifying a non-existent font at build
+ * time.
+ */
+function createFont<
+  RawFont extends { value: RawFontValue; label: RawFontLabel },
```

Comment:
I really like how this PR is laden with improvement examples. Not suggesting we do it for this PR: the introduction of the storybook, and the individual component improvements (eg. this advanced font handling, or the `commit` improvement) would deserve separate PRs to make them separately referenceable and digestible, as the commit train gets squashed on merge to `master`.

Reply:
Agreed.  I was being holistic with this PR.